errbot/backends/slack.py

```diff
@@ -531,6 +532,12 @@ def _message_event_handler(self, event):
         if mentioned:
             self.callback_mention(msg, mentioned)
 
+    def _member_joined_channel_event_handler(self, event):
+        """Event handler for the 'member_joined_channel' event"""
+        user = event['user']
+        if user == self.bot_identifier:
```

Comment:
does it work?
`user` is a string and `self.bot_identifier` is a SlackPerson. Maybe reconstruct the SlackPerson like for the presence event handler and compare? `SlackPerson(self.sc, event['user'])`

Reply:
Good catch!  I tested that it didn't kick off when a person joined the channel but forgot to also check that it worked when the bot joined the channel.  I've updated the code and tested both instances and it's working as expected.  Thank you!